src/include/storage/write_ahead_log/log_manager.h

```diff
@@ -68,25 +98,112 @@ class LogManager {
   void Flush();
 
  private:
+  friend class LogConsumerTask;
+
+  bool run_log_manager_;
+
   // TODO(Tianyu): This can be changed later to be include things that are not necessarily backed by a disk
-  // (e.g. logs can be streamed out to the network for remote replication)
-  BufferedLogWriter out_;
+  //  (e.g. logs can be streamed out to the network for remote replication)
   RecordBufferSegmentPool *buffer_pool_;
 
   // TODO(Tianyu): Might not be necessary, since commit on txn manager is already protected with a latch
   common::SpinLatch flush_queue_latch_;
   // TODO(Tianyu): benchmark for if these should be concurrent data structures, and if we should apply the same
-  // optimization we applied to the GC queue.
+  //  optimization we applied to the GC queue.
   std::queue<RecordBufferSegment *> flush_queue_;
 
   // These do not need to be thread safe since the only thread adding or removing from it is the flushing thread
   std::vector<std::pair<transaction::callback_fn, void *>> commits_in_buffer_;
 
+  // System path for log file
+  const char *log_file_path_;
```

Comment:
Probably needs to be lifted up to settings manager

Reply:
Done.